pkg/nix/cmd/build.go

```diff
@@ -13,10 +16,30 @@ func Build(dir string, attribute string) error {
 	}
 	cmd := exec.Command("nix", "build", attribute, "-o", dir)
 
-	cmd.Stdout = os.Stdout
-	// TODO: in future- we can pipe to stderr pipe and modify error messages to be understandable by the user
-	cmd.Stderr = os.Stderr
-
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		log.Fatalf("could not get stderr pipe: %v", err)
```

Comment:
As a golang convention, we shouldn't Fatal/os.Exit(1) in `pkg/` . Return the error and handle it in `cmd/`